packages/components/src/disabled/index.js

```diff
@@ -52,31 +53,54 @@ class Disabled extends Component {
 			attributes: true,
 			subtree: true,
 		} );
+
+		this.node.addEventListener( 'click', this.catchEvents );
+		this.node.addEventListener( 'focus', this.catchEvents );
 	}
 
 	componentWillUnmount() {
 		this.observer.disconnect();
 		this.debouncedDisable.cancel();
+		this.node.removeEventListener( 'click', this.catchEvents );
+		this.node.removeEventListener( 'focus', this.catchEvents );
+	}
+
+	catchEvents( e ) {
+		e.preventDefault();
+		e.stopPropagation();
 	}
 
 	bindNode( node ) {
 		this.node = node;
 	}
 
 	disable() {
-		focus.focusable.find( this.node ).forEach( ( focusable ) => {
-			if ( includes( DISABLED_ELIGIBLE_NODE_NAMES, focusable.nodeName ) ) {
-				focusable.setAttribute( 'disabled', '' );
-			}
+		const {
+			onDisable = noop,
+			eligibleNodeNames = DISABLED_ELIGIBLE_NODE_NAMES,
+		} = this.props;
+
+		const focusableNodes = focus.focusable.find( this.node );
 
-			if ( focusable.hasAttribute( 'tabindex' ) ) {
-				focusable.removeAttribute( 'tabindex' );
+		focusableNodes.forEach( ( focusable ) => {
+			if ( includes( eligibleNodeNames, focusable.nodeName ) ) {
+				focusable.setAttribute( 'disabled', '' );
 			}
 
 			if ( focusable.hasAttribute( 'contenteditable' ) ) {
 				focusable.setAttribute( 'contenteditable', 'false' );
 			}
+
+			// Make it impossible to focus or tab to `<a>` or `<area>` els
+			if ( focusable.hasAttribute( 'href' ) ) {
+				focusable.removeAttribute( 'href' );
+			}
+
+			// Disable all nodes from being tabbable order
+			focusable.setAttribute( 'tabindex', '-1' );
 		} );
+
+		onDisable( focusableNodes );
```

Comment:
Should we remove ` onDisable` prop and update the README?

Reply:
✅  Done and done.